srsly/_json_api.py

```diff
@@ -53,6 +79,42 @@ def read_json(path: FilePath) -> JSONOutput:
         return ujson.load(f)
 
 
+def read_json_dict(path: FilePath) -> Dict[str, Any]:
+    """Load JSON from file or standard input.
+
+    path (FilePath): The file path. "-" for reading from stdin.
+    RETURNS (JSONOutput): The loaded JSON content.
+    """
+    data = read_json(path)
+    if not isinstance(data, dict):
+        raise ValueError("Invalid JSON, data could not be parsed to a dict.")
+    return data
+
+
+def read_json_list(path: FilePath, validate_inner: bool = False, skip_invalid: bool = False) -> List[Dict[str, Any]]:
```

Comment:
This is still a bit of a typing problem because the return type depends on the value of `validate_inner`.

I think this method with these options and types may be getting too specific for inclusion in the `srsly` API.

If it is going to be in `srsly`, then I think it would be better as `read_json_list() -> List` and `read_json_list_of_dicts() -> List[Dict[str, JSONOutput]]` or something along those lines?

Reply:
Agreed that was weird. I fixed these types and separated this logic into 2 functions.

This parsing as a list idea is kinda extra honestly. I think the main functions I really think should live in srsly are:

* `read_json_dict` -> We parse a normal JSON file expecting a JSON object (e.g. for configs and stuff) all the time
* `read_jsonl_dicts` -> We use this to load data all the time and expect each line to be a valid JSON object